### PR TITLE
Causelist title and end date

### PR DESCRIPTION
**Pull request by actlikewill**
image

closes https://github.com/laws-africa/kenyalaw-pj/issues/256
closes https://github.com/laws-africa/peachjam/issues/2117

### CODE REVIEW

File: peachjam/models/judgment.py [hunk 1]
```diff
@@ -583,15 +583,20 @@ class CauseList(CoreDocument):
...
 
     def assign_title(self):
-        court_name = self.registry.name if self.registry else ""
-        if not self.registry:
-            court_name = self.court.name if self.court else ""
-        nature_name = self.nature.name if self.nature else ""
-        date_str = self.date.strftime("%d %B %Y")
-
-        self.title = f"{court_name} - {nature_name} - {date_str}"
+        if not self.title:
```
Comment (longhotsummer) [inline peachjam/models/judgment.py:589]: surely we want to update the title if any details change?

File: peachjam/models/judgment.py [hunk 2]
```diff
@@ -583,15 +583,20 @@ class CauseList(CoreDocument):
...
+            court_name = f"{self.registry.name} –" if self.registry else ""
+            if not self.registry:
+                court_name = f" {self.court.name} –" if self.court else ""
+            division = f" {self.division.name} Division –" if self.division else ""
+            nature_name = f" {self.nature.name} –" if self.nature else ""
+            date_str = f' {self.date.strftime("%d %B %Y")}'
+            end_date = (
+                f' to {self.end_date.strftime("%d %B %Y")}' if self.end_date else ""
+            )
+            self.title = f"{court_name}{division}{nature_name}{date_str}{end_date}"
```
Comment (longhotsummer) [inline peachjam/models/judgment.py:599]: Sometime's it's simpler to put all the elements into an array and do `self.title = ' - '.join(elements)`

File: peachjam/models/judgment.py (hunk 1 above)
Comment (actlikewill) [reply peachjam/models/judgment.py]: I was thinking that an editor might want to change the title but I see that its incorrect

**Review by longhotsummer — APPROVED** [review]

File: peachjam/models/judgment.py (hunk 2 above)
Comment (actlikewill) [reply peachjam/models/judgment.py]: agreed, much better now

**Pull request merged**